language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -221,7 +229,8 @@ computeCek con t@(Var _ varName)   = do
 returnCek
     :: (GShow uni, GEq uni, DefaultUni <: uni, Closed uni, uni `Everywhere` ExMemoryUsage)
     => Context uni -> WithMemory Value uni -> CekM uni (Plain Term uni)
-returnCek [] res = pure $ void res
+-- Instantiate all the free variable of the resulting term in case there are any.
```

Comment:
:+1:

Reply:
This would only be possible if someone actually started the original `computeCek` with a non-empty environment, right?

Reply:
No. Thanks for the reminder. The lack of "environment discharging" is the reason why we still use the CK machine in `PlcTestUtils`. Otherwise the `sameEmptyRoseEval` test of `plutus-tx-plugin` fails. Basically if evaluation results in a non-constant, then that non-constant can have uninstantiated variables in it. We've discussed that somewhat recently, but I'm unable to find the discussion.

I'll update the `PlcTestUtils` module to finally use the CEK machine.

Reply:
Sounds good, although we should have some tests to cross-check them somewhere.

Reply:
Added a test and checked that the behaviour of the CEK machine has indeed changed.